metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/common/helpers/MacrosUpdater.java

```diff
@@ -0,0 +1,568 @@
+	static final String BASIC_AUTH_BASE64_ESC_POWERSHELL = "%{BASIC_AUTH_BASE64_ESC_CMD}";
+	static final String SHA256_AUTH_ESC_POWERSHELL = "%{SHA256_AUTH_ESC_CMD}";
+	static final String PASSWORD_BASE64_ESC_POWERSHELL = "%{PASSWORD_BASE64_ESC_CMD}";
+
+	// Linux Shell input format macros
+	static final String PASSWORD_ESC_LINUX = "%{PASSWORD_ESC_LINUX}";
+	static final String USERNAME_ESC_LINUX = "%{USERNAME_ESC_LINUX}";
+	static final String HOSTNAME_ESC_LINUX = "%{HOSTNAME_ESC_LINUX}";
+	static final String AUTHENTICATIONTOKEN_ESC_LINUX = "%{AUTHENTICATIONTOKEN_ESC_LINUX}";
+	static final String BASIC_AUTH_BASE64_ESC_LINUX = "%{BASIC_AUTH_BASE64_ESC_LINUX}";
+	static final String SHA256_AUTH_ESC_LINUX = "%{SHA256_AUTH_ESC_LINUX}";
+	static final String PASSWORD_BASE64_ESC_LINUX = "%{PASSWORD_BASE64_ESC_LINUX}";
+
+	// Bash input format macros (Linux Shell alias)
+	static final String PASSWORD_ESC_BASH = "%{PASSWORD_ESC_BASH}";
+	static final String USERNAME_ESC_BASH = "%{USERNAME_ESC_BASH}";
+	static final String HOSTNAME_ESC_BASH = "%{HOSTNAME_ESC_BASH}";
+	static final String AUTHENTICATIONTOKEN_ESC_BASH = "%{AUTHENTICATIONTOKEN_ESC_BASH}";
+	static final String BASIC_AUTH_BASE64_ESC_BASH = "%{BASIC_AUTH_BASE64_ESC_BASH}";
+	static final String SHA256_AUTH_ESC_BASH = "%{SHA256_AUTH_ESC_BASH}";
+	static final String PASSWORD_BASE64_ESC_BASH = "%{PASSWORD_BASE64_ESC_BASH}";
+
+	// SQL input format macros
+	static final String PASSWORD_ESC_SQL = "%{PASSWORD_ESC_SQL}";
+	static final String USERNAME_ESC_SQL = "%{USERNAME_ESC_SQL}";
+	static final String HOSTNAME_ESC_SQL = "%{HOSTNAME_ESC_SQL}";
+	static final String AUTHENTICATIONTOKEN_ESC_SQL = "%{AUTHENTICATIONTOKEN_ESC_SQL}";
+	static final String BASIC_AUTH_BASE64_ESC_SQL = "%{BASIC_AUTH_BASE64_ESC_SQL}";
+	static final String SHA256_AUTH_ESC_SQL = "%{SHA256_AUTH_ESC_SQL}";
+	static final String PASSWORD_BASE64_ESC_SQL = "%{PASSWORD_BASE64_ESC_SQL}";
+
+	/**
+	 * Replaces each known HTTP macro in the given text with the literal target sequences:<br>
+	 * username, password, authentication-token, base64-password, base64-auth and sha256-auth
+	 *
+	 * @param text                The text we wish to update
+	 * @param username            The HTTP username
+	 * @param password            The HTTP password
+	 * @param authenticationToken The HTTP Authentication Token
+	 * @param hostname            The remote hostname
+	 * @return String value
+	 */
+	public static String update(
+		String text,
+		String username,
+		char[] password,
+		String authenticationToken,
+		@NonNull String hostname
+	) {
+		if (text == null || text.isEmpty()) {
+			return EMPTY;
+		}
+
+		// Null values control
+		final String passwordAsString = password != null ? String.valueOf(password) : EMPTY;
+		username = username != null ? username : EMPTY;
+		authenticationToken = authenticationToken != null ? authenticationToken : EMPTY;
+
+		// Replace provided macros which don't need processing
+		String updatedContent = text
```

Comment:
Before running this giant chain of `replace()` calls, we should check whether `text` contains `%{`, to optimize things a little bit.

Also, I thought we would be searching for a regular expression like: `%\{(\w+?)(?:_ESC_(\w+))?\}`, where:

* `group(1)` = macro name (e.g. `USERNAME`)
* `group(2)` = escaping (e.g. `JSON`, or `null` if not present)

Looping through matches of the above regular expression will be more extensible, and maybe even better in terms of performance.

Final thought: we will need this escape mechanism for other references, like `%{var::...}`, `%{source::...}`, etc.. We need to think of the best syntax to handle these.